{{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py

```diff
@@ -10,56 +10,72 @@
 {%- set widget_name = cookiecutter.plugin_name.replace(' ', '') + 'Widget' -%}
 {%- set config_page = cookiecutter.plugin_name.replace(' ', '') + 'ConfigPage' %}
 
+{%- if cookiecutter.graphical_plugin != 'n' %}
 from qtpy.QtWidgets import QVBoxLayout
-
-{% if cookiecutter.spyder3_compatibility == 'y' -%}
+{%- endif %}
+{% if cookiecutter.spyder3_compatibility == 'y' %}
 try:
     from spyder.api.plugins import SpyderPluginWidget
 except ImportError:
     from spyder.plugins import SpyderPluginWidget # Spyder3
     {%- if cookiecutter.create_config_page == 'y' %}
 try:
-    from spyder.api.preferences import
+    from spyder.api.preferences import PluginConfigPage
 except ImportError:
     from spyder.plugins.configdialog import PluginConfigPage # Spyder3
     {%- endif %}
-{% else -%}
+{% else %}
+    {%- if cookiecutter.graphical_plugin == 'y' %}
 from spyder.api.plugins import SpyderPluginWidget
+    {%- else %}
+from spyder.api.plugins import SpyderPlugin
+    {%- endif %}
     {%- if cookiecutter.create_config_page == 'y' %}
 from spyder.api.preferences import PluginConfigPage
-    {%- endif %}
-{%- endif -%}
-
+    {% endif %}
+{%- endif %}
 from .widgets.{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui import {{ widget_name }}
-{% if cookiecutter.create_config_page == 'y' %}
 
+{% if cookiecutter.create_config_page == 'y' %}
 class {{ config_page }}(PluginConfigPage):
     """{{ cookiecutter.plugin_name }} plugin preferences."""
     pass
 {% endif %}
-
+{%- if cookiecutter.graphical_plugin == 'n' %}
+class {{ cookiecutter.plugin_name.replace(' ', '') }}Plugin(SpyderPlugin):
+    """{{ cookiecutter.plugin_name }} plugin."""
+    CONF_SECTION = '{{ cookiecutter.plugin_name }}'
```

Comment:
Another thing I just noticed: this needs to be all in lowercase letters, i.e. if the name of the plugin is `Demo`, this needs to be `demo`.